lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -1435,6 +1443,11 @@ namespace {
         if (!SP) {
           // If there's no sub-pattern then there's no further recursive
           // structure here.  Yield the constructor space.
+          if(VP->hasUnresolvedOriginalExpr()){
```

Comment:
The space engine does not need to concern itself with ill-formed pattern trees.  They will have been structurally diagnosed by the rest of pattern type checking, and we should not project them.